### PR TITLE
[Improvement](topn) add limit threashold session variable and fuzzy for topn optimizations

### PR DESCRIPTION
**Pull request by xiaokang**
# Proposed changes

Issue Number: close #xxx

## Problem summary

1. add limit threshold for topn runtime pushdown and key topn optimization
2. use unified session variable topn_opt_limit_threshold for all topn optimizations
3. add fuzzy support for topn_opt_limit_threshold

## Checklist(Required)

1. Does it affect the original behavior: 
    - [ ] Yes
    - [x] No
    - [ ] I don't know
4. Has unit tests been added:
    - [ ] Yes
    - [ ] No
    - [x] No Need
5. Has document been added or modified:
    - [x] Yes
    - [ ] No
    - [ ] No Need
6. Does it need to update dependencies:
    - [ ] Yes
    - [x] No
7. Are there any changes that cannot be rolled back:
    - [ ] Yes (If Yes, please explain WHY)
    - [x] No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by eldenmoon** [comment]
we could add this limit to fuzzy sesion variable

**Comment by xiaokang** [comment]
> we could add this limit to fuzzy sesion variable

added

**Review by eldenmoon — DISMISSED** [review]
LGTM

File: fe/fe-core/src/main/java/org/apache/doris/planner/OriginalPlanner.java [hunk 1]
```diff
@@ -448,6 +448,7 @@ private void checkAndSetTopnOpt(PlanNode node) {
             SortNode sortNode = (SortNode) node;
             PlanNode child = sortNode.getChild(0);
             if (child instanceof OlapScanNode && sortNode.getLimit() > 0
+                    && sortNode.getLimit() <= ConnectContext.get().getSessionVariable().topnOptLimitThreshold
```
Comment (Gabriel39) [inline fe/fe-core/src/main/java/org/apache/doris/planner/OriginalPlanner.java:451]: I'm not sure if session will be null here

Comment (xiaokang) [reply fe/fe-core/src/main/java/org/apache/doris/planner/OriginalPlanner.java:452]: added check for null

**Review by Gabriel39 — APPROVED** [review]
LGTM

**Pull request merged**